### PR TITLE
Bolt address with same origin as http address

### PR DESCRIPTION
**Pull request by maxsumrall**
SharedServerTestBase should ultimately not have those changes.

### CODE REVIEW

File: community/server-api/src/main/java/org/neo4j/server/rest/repr/Serializer.java [hunk 1]
```diff
@@ -76,6 +76,11 @@ final void serialize( ListWriter list, ListRepresentation value )
         list.done();
     }
 
+    final String relativeBoltUri(String boltPort){
+        String bolt = baseUri.toString().replaceFirst( baseUri.getScheme(), "bolt" );
```
Comment (apcj) [inline community/server-api/src/main/java/org/neo4j/server/rest/repr/Serializer.java]: This seems a bit backward to me. Why throw away the parsed URI and use string manipulation (replaceFirst, length etc.) when when you can neatly extract the host part out of the URI with a method call?

File: community/server/src/test/java/org/neo4j/server/rest/DiscoveryServiceIT.java [hunk 2]
```diff
@@ -83,14 +83,39 @@ public void shouldRedirectOnHtmlRequest() throws Exception
...
+        // given
+        RestRequest request = new RestRequest( server().baseUri() ).host( "boltprotocol.org" );
+
+        // when
+        JaxRsResponse response = request.get();
+
+        // then
+        Map<String,Object> map = JsonHelper.jsonToMap( response.getEntity() );
+        assertThat( String.valueOf( map.get( "bolt" ) ),
+                containsString( "bolt://" + request.getHeaders().get("host") ) );
```
Comment (apcj) [inline community/server/src/test/java/org/neo4j/server/rest/DiscoveryServiceIT.java]: This doesn't seem like a very strong assertion. For example, does the URI have the correct port?

Comment (maxsumrall) [reply community/server/src/test/java/org/neo4j/server/rest/DiscoveryServiceIT.java]: Yes I'll amend it to also check for the port. The main concern was to test that the host part of the address was correct.

File: community/server-api/src/main/java/org/neo4j/server/rest/repr/Serializer.java (hunk 1 above)
Comment (maxsumrall) [reply community/server-api/src/main/java/org/neo4j/server/rest/repr/Serializer.java]: Yes, I had also found that to be a better way to do this.

**Comment by chrisvest** [comment]
@maxsumrall Looks like some extra commits snuck in by accident somehow.

**Comment by jimwebber** [comment]
Following some conversation with @benbc we're going to affine this with the existing Bolt advertised address stuff such that an explicitly configured address has precedence over an inferred address. Until that's pushed, this remains not ready for merge.

**Comment by benbc** [comment]
Fine by me.

**Pull request merged**